ecs/pod_creator.go

```diff
@@ -27,11 +31,227 @@ func NewBasicECSPodCreator(c cocoa.ECSClient, v cocoa.Vault) (*BasicECSPodCreato
 
 // CreatePod creates a new pod backed by AWS ECS.
 func (m *BasicECSPodCreator) CreatePod(ctx context.Context, opts ...*cocoa.ECSPodCreationOptions) (cocoa.ECSPod, error) {
-	return nil, errors.New("TODO: implement")
+
+	mergedPodCreationOpts := cocoa.MergeECSPodCreationOptions(opts...)
+	mergedPodExecutionOpts := cocoa.MergeECSPodExecutionOptions(mergedPodCreationOpts.ExecutionOpts)
+
+	err := mergedPodCreationOpts.Validate()
+	if err != nil {
+		return nil, errors.Wrap(err, "invalid pod creation options")
+	}
+
+	err = mergedPodExecutionOpts.Validate()
+	if err != nil {
+		return nil, errors.Wrap(err, "invalid pod execution options")
+	}
+
+	allEnvVars, allSecrets, allSecretsToCreate, err := m.exportSecrets(ctx, mergedPodCreationOpts)
+	if err != nil {
+		return nil, errors.Wrap(err, "exporting secrets")
+	}
+
+	err = m.createSecrets(ctx, allSecretsToCreate)
+	if err != nil {
+		return nil, errors.Wrap(err, "creating secrets")
+	}
+
+	taskDefinition, err := m.exportPodCreationOptions(ctx, mergedPodCreationOpts, allEnvVars, allSecrets)
+	if err != nil {
+		return nil, errors.Wrap(err, "translating the register task definition input to the correct type")
+	}
+
+	registerOut, err := m.client.RegisterTaskDefinition(ctx, taskDefinition)
+	if err != nil {
+		return nil, errors.Wrap(err, "registering task definition")
+	}
+
+	if registerOut.TaskDefinition == nil || registerOut.TaskDefinition.TaskDefinitionArn == nil {
+		return nil, errors.New("missing task definition")
+	}
+
+	taskDef := cocoa.NewECSTaskDefinition()
+	taskDef = taskDef.SetID(*registerOut.TaskDefinition.TaskDefinitionArn)
+	taskDef = taskDef.SetOwned(*utility.TruePtr())
+
+	runTask := &ecs.RunTaskInput{}
+	runTask.SetCluster(*mergedPodExecutionOpts.Cluster).
+		SetTaskDefinition(*taskDef.ID).
+		SetTags(registerOut.Tags).
```

Comment:
Use the tags from the `mergedPodExecutionOpts.Tags`, not the register output. The tags that you set when you register a task definition are distinct and not necessarily related to the ones for the actual running task (because they're considered separate resources in ECS).